cmd/clusterctl/cmd/generate_yaml.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client"
+)
+
+type generateYAMLOptions struct {
+	url           string
+	listVariables bool
+}
+
+var gyOpts = &generateYAMLOptions{}
+
+var generateYamlCmd = &cobra.Command{
+	Use:   "yaml",
+	Short: "Process yaml using clusterctl's yaml processor",
+	Long: LongDesc(`
+		Process yaml using clusterctl's yaml processor.
+
+		clusterctl ships with a simple yaml processor that performs variable
+		substitution that takes into account of default values.
+
+		Variable values are either sourced from the clusterctl config file or
+		from environment variables`),
+
+	Example: Examples(`
+		# Generates a configuration file with variable values using
+		a template from a specific URL.
+		clusterctl generate yaml --from https://github.com/foo-org/foo-repository/blob/master/cluster-template.yaml
+
+		# Generates a configuration file with variable values using
+		a template stored locally.
+		clusterctl generate yaml  --from ~/workspace/cluster-template.yaml`),
```

Comment:
Do users have to specify the `--from` command at any given time, or could it default to stdin? I'm wondering because I can see some use cases where you'd want the template to be piped through multiple commands first and then to clusterctl

Reply:
Hmm...that makes sense. Currently, it doesn't read from stdin. I used the `--from` flag since it is also used in `clusterctl config cluster` and would provide some sense of consistency between commands. 

We could do something like `cat my-template | clusterctl generate yaml --from -`. This would emulate `kubectl apply  --filename -`. 
Maybe we can do that as part of a separate PR. WDYT?